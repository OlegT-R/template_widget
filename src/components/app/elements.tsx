import * as styledComponents from 'styled-components';

const { default: styled } = styledComponents;

export const AppBox = styled.div`
  margin-top: 20px;
`;
