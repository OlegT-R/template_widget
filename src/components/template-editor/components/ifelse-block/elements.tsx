import { Button } from 'antd';
import * as styledComponents from 'styled-components';

const { default: styled } = styledComponents;

export const Block = styled.div`
  width: 100%;
`;

export const BlockLine = styled.div`
  width: 100%;
  padding: 5px 0;
  display: flex;
  align-items: flex-start;
`;

export const OperatorCell = styled.div`
  width: 150px;
`;

export const ContentCell = styled.div`
  width: calc(100% - 150px);
`;

export const StyledBtn = styled(Button)`
  margin-left: 15px;
`;
