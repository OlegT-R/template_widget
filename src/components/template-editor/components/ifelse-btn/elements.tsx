import * as styledComponents from 'styled-components';

const { default: styled } = styledComponents;

export const Operator = styled.span`
  background: #1890ff;
  padding: 2px 3px;
  margin: 0 2px;
  color: #fff;
  font-size: 14px;
  font-weight: bold;
  border-radius: 5px;
  text-transform: uppercase;
`;
