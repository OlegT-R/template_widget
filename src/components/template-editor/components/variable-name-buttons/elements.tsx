import { Button } from 'antd';
import styled from 'styled-components';

export const VarBtn = styled(Button)`
    margin-right: 5px;
`;

export const BtnsBox = styled.div`
  position: relative;
  width: 100%;
  display: flex;
`;
