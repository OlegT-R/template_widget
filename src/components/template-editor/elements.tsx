import { Button } from 'antd';
import * as styledComponents from 'styled-components';

const { default: styled } = styledComponents;

export const EditorContainer = styled.div`
  width: 800px;
  height: auto;
  min-height: 300px;
  margin: 0 auto;
  position: relative;
`;

export const IfElseBox = styled.div`
  text-align: center;
  margin-top: 15px;
`;

export const TemplateBox = styled.div`
  width: 100%;
  overflow-x: auto;
  margin: 20px 0 10px 0;
`;

export const StyledBtn = styled(Button)`
  margin-right: 15px;
`;
