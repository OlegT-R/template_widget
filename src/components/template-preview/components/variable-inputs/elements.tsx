import { Input } from 'antd';
import styled from 'styled-components';

export const VarInput = styled(Input)`
  margin: 5px 5px 0 0;
  width: 200px;
`;

export const Label = styled.span`
  padding-right: 5px;
  font-size: 16px;
  font-weight: bold;
`;

export const BtnsBox = styled.div`
  padding: 10px 0;
  position: relative;
  width: 100%;
  display: flex;
  margin-top: 10px;
  flex-wrap: wrap;
  align-items: center;
`;
