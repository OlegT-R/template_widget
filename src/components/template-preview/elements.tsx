import { Input } from 'antd';
import * as styledComponents from 'styled-components';

const { default: styled } = styledComponents;
const { TextArea } = Input;

export const PreviewBlock = styled.div`
  width: 100%;
  max-width: 600px;
  height: auto;
  min-height: 200px;
  margin: 0 auto;
  position: relative;
`;
export const StyledArea = styled(TextArea)`
  resize: none;
`;
