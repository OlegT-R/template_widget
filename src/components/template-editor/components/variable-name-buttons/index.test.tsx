import React from 'react';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { VariableNameBtns } from '.';
import { VarBtn } from './elements';

const variables = ['var1', 'var2', 'var3'];

test('<VariableNameBtns />', () => {
  const component = shallow(
    <VariableNameBtns arrVarNames={variables} onBtnClick={() => null} />
  );
  expect(component.find(VarBtn).length).toBe(3);
  expect(toJson(component)).toMatchSnapshot();
});
