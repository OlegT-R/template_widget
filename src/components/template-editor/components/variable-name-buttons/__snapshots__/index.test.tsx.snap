// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VariableNameBtns /> 1`] = `
<styled.div>
  <Styled(Button)
    key="var1"
    onClick={[Function]}
    type="primary"
  >
    {var1}
  </Styled(Button)>
  <Styled(Button)
    key="var2"
    onClick={[Function]}
    type="primary"
  >
    {var2}
  </Styled(Button)>
  <Styled(Button)
    key="var3"
    onClick={[Function]}
    type="primary"
  >
    {var3}
  </Styled(Button)>
</styled.div>
`;
